insonmnia/miner/server.go

```diff
@@ -371,6 +373,40 @@ func (m *Miner) sendUpdatesOnRequest(server pb.Miner_TasksStatusServer) {
 }
 
 // TasksStatus returns the status of a task
+func (m *Miner) TaskLogs(request *pb.TaskLogsRequest, server pb.Miner_TaskLogsServer) error {
+	log.G(m.ctx).Info("handling TaskLogs request", zap.Any("request", request))
+	cid, ok := m.getContainerIdByTaskId(request.Id)
+	if !ok {
+		return status.Errorf(codes.NotFound, "no job with id %s", request.Id)
+	}
+	opts := types.ContainerLogsOptions{
+		ShowStdout: request.Type == pb.TaskLogsRequest_STDOUT || request.Type == pb.TaskLogsRequest_BOTH,
+		ShowStderr: request.Type == pb.TaskLogsRequest_STDERR || request.Type == pb.TaskLogsRequest_BOTH,
+		Since:      request.Since,
+		Timestamps: request.AddTimestamps,
+		Follow:     request.Follow,
+		Tail:       request.Tail,
+		Details:    request.Details,
+	}
+	reader, err := m.ovs.Logs(server.Context(), cid, opts)
+	if err != nil {
+		return err
+	}
+	buffer := make([]byte, 100*1024)
+	for {
+		readCnt, err := reader.Read(buffer)
+		if readCnt != 0 {
+			server.Send(&pb.TaskLogsChunk{string(buffer[:readCnt])})
+		}
+		if err == io.EOF {
+			return nil
+		} else if err != nil {
```

Comment:
if err != nil is sacred in go! It's better to do same across the code. Look at hub/server.go +210